script.js

```diff
@@ -0,0 +1,690 @@
+  // get a random index from an array given it's size
+  const getRandomIndex = (size) => Math.floor(Math.random() * size);
+  // loop over the entire cards array
+  for (let currentIndex = 0; currentIndex < cards.length; currentIndex += 1) {
+    // select a random position from the deck
+    const randomIndex = getRandomIndex(cards.length);
+
+    // get the current card in the loop
+    const currentItem = cards[currentIndex];
+
+    // get the random card
+    const randomItem = cards[randomIndex];
+
+    // swap the current card and the random card
+    cards[currentIndex] = randomItem;
+    cards[randomIndex] = currentItem;
+  }
+
+  // give back the shuffled deck
+  return cards;
+};
+const makeDeck = () => {
+  // create the empty deck at the beginning
+  const newDeck = [];
+  const suits = ['hearts', 'diamonds', 'clubs', 'spades'];
+
+  for (let suitIndex = 0; suitIndex < suits.length; suitIndex += 1) {
+    // make a variable of the current suit
+    const currentSuit = suits[suitIndex];
+    let colorType;
+    let suitSymbol;
+    if (suitIndex === 0) {
+      suitSymbol = '♥';
+      colorType = 'red';
+    } else if (suitIndex === 1) {
+      suitSymbol = '♦';
+      colorType = 'red';
+    } else if (suitIndex === 2) {
+      suitSymbol = '♣';
+      colorType = 'black';
+    } else {
+      suitSymbol = '♠';
+      colorType = 'black';
+    }
+
+    // loop to create all cards in this suit
+    // rank 1-13
+    for (let rankCounter = 1; rankCounter <= 13; rankCounter += 1) {
+      // Convert rankCounter to string
+      let cardName = `${rankCounter}`;
+      let displayName;
+
+      // 1, 11, 12 ,13
+      if (cardName === '1') {
+        cardName = 'A';
+      } else if (cardName === '11') {
+        cardName = 'J';
+      } else if (cardName === '12') {
+        cardName = 'Q';
+      } else if (cardName === '13') {
+        cardName = 'K';
+      }
+      if (cardName === 'A') {
+        displayName = 'A';
+      } else if (cardName === 'J') {
+        displayName = 'J';
+      } else if (cardName === 'Q') {
+        displayName = 'Q';
+      } else if (cardName === 'K') {
+        displayName = 'K';
+      } else {
+        displayName = cardName;
+      }
+
+      // make a single card object variable
+      const cardInfo = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+        symbol: suitSymbol,
+        color: colorType,
+        display: displayName,
+      };
+
+      // add the card to the deck
+      newDeck.push(cardInfo);
+    }
+  }
+
+  return newDeck;
+};
+// make deck
+let deck = shuffleCards(makeDeck());
+
+const loadDeck = () => {
+  deck = shuffleCards(makeDeck());
+};
+/* <==== Game element we need / what the user sees ===>
+1. Username input box (Done)
+2. Submit button (done)
+3. Bid amount input box (done)
+4. Display of 5 Cards + Card Container (done)
+5. Winnings shown on screen (done)
+6. Click to deal button (done)
+*/
+
+// Creating the elements on screen
```

Comment:
Consider declaring static HTML elements directly in the HTML code to minimise the logic needed in your JS files.

Reply:
I thought we had to create everything via DOM.. That's why I didn't input anything into the index.html file